src/cljs/kefirnadar/application/views.cljs

```diff
@@ -195,56 +23,69 @@
 
 
 (defn first-name-input [id]
-  (let [value (subscribe [::subs/form id])]
-    [:div
-     [:label "Ime"]
+  (let [value (subscribe [::subs/form id])
+        valid? (subscribe [::subs/form-validation id])]
+    [:div.d-flex.flex-column.justify-content-center.align-items-center.mb-3
+     [:label "Ime: "]
      [:input {:value       @value
               :on-change   #(dispatch [::events/update-form id (extract-input-value %)])
+              :on-blur     #(dispatch [::events/validate-form id (validation/field-validation id (extract-input-value %))])
               :type        "text"
-              :placeholder "Vase ime..."}]]))
+              :required    true
+              :placeholder "Vase ime..."}]
+     (if (not @valid?) [:p.text-danger "  Unesite vase ime."])]))
 
 
 (defn last-name-input [id]
-  (let [value (subscribe [::subs/form id])]
-    [:div
-     [:label "Prezime"]
+  (let [value (subscribe [::subs/form id])
+        valid? @(subscribe [::subs/form-validation id])]
+    [:div.d-flex.flex-column.justify-content-center.align-items-center.mb-3
+     [:label "Prezime: "]
      [:input {:value       @value
               :on-change   #(dispatch [::events/update-form id (extract-input-value %)])
+              :on-blur     #(dispatch [::events/validate-form id (validation/field-validation id (extract-input-value %))])
               :type        "text"
-              :placeholder "Vase prezime..."}]]))
+              :placeholder "Vase prezime..."}]
+     (if (not valid?) [:p.text-danger "  Unesite vase prezime."])]))
 
 
 (defn region-select [id regions]
   (let [value (subscribe [::subs/form id])]
-    [:div
-     [:label " Opstina: "]
-     [:div
-      [:select {:value     @value
-                :on-change #(dispatch [::events/update-form id (keyword (extract-input-value %))])}
-       [:option {:value ""} "Izabarite opstinu"]
-       (map (fn [r] [:option {:key r :value r} r]) regions)]]]))
+    [:div.d-flex.flex-column.justify-content-center.align-items-center.mb-3
+     [:label "Opstina:"]
+     [:select {:value     @value
+               :on-change #(dispatch [::events/update-form id (keyword (extract-input-value %))])}
+      [:option {:value ""} "Izabarite opstinu"]
+      (map (fn [r] [:option {:key r :value r} r]) regions)]]))
+
 
 (defn email-input [id]
-  (let [value (subscribe [::subs/form id])]
-    [:div
-     [:label "E-mail"]
+  (let [value (subscribe [::subs/form id])
+        valid? @(subscribe [::subs/form-validation id])]
+    [:div.d-flex.flex-column.justify-content-center.align-items-center.mb-3
+     [:label "E-mail: "]
      [:input {:value       @value
               :on-change   #(dispatch [::events/update-form id (extract-input-value %)])
+              :on-blur     #(dispatch [::events/validate-form id (validation/field-validation id (extract-input-value %))])
               :type        "text"
-              :placeholder "Vasa elektronska posta..."}]]))
+              :placeholder "xxxx@xxxx.com"}]
```

Comment:
ne mora da bude `.com`. može i umesto toga `xxx`.

Reply:
Promenjeno, sto se tice tog frontend-a meni se ne svidja i videcu to da promenim, zapravo dosta stvari mi se ne svidja tu, znam da to mogu mnogo bolje da uradim